### PR TITLE
MULE-13534: Application lazy initialization should allow to enable mo…

### PR DESCRIPTION
**Pull request by gsfernandes**
…re than one component

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/api/LazyComponentInitializer.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.spring.api;
```
Comment (pablolagreca) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/api/LazyComponentInitializer.java]: we can't have api with a package spring

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/MinimalApplicationModelGenerator.java [hunk 2]
```diff
@@ -45,14 +43,45 @@ public MinimalApplicationModelGenerator(ConfigurationDependencyResolver dependen
...
+   * @param filter to select the {@link ComponentModel componentModels} to be enabled.
+   * @return the generated {@link ApplicationModel} with the minimal set of {@link ComponentModel}s required.
+   */
+  public ApplicationModel getMinimalModel(LazyComponentInitializer.ComponentLocationFilter filter) {
+    List<ComponentModel> required = dependencyResolver.findRequiredComponentModels(filter);
+    required.stream().forEach(componentModel -> {
+      final DefaultComponentLocation componentLocation = componentModel.getComponentLocation();
+      if (componentLocation != null) {
+        final String location = componentLocation.getLocation();
+        //TODO (gfernandes) munit components has no location :S
```
Comment (pablolagreca) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/MinimalApplicationModelGenerator.java]: We need to review ComponentLocatorVisitor then

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/api/LazyComponentInitializer.java (hunk 1 above)
Comment (gsfernandes) [reply modules/spring-config/src/main/java/org/mule/runtime/config/spring/api/LazyComponentInitializer.java]: https://www.mulesoft.org/jira/browse/MULE-13592

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/MinimalApplicationModelGenerator.java (hunk 2 above)
Comment (gsfernandes) [reply modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/MinimalApplicationModelGenerator.java]: Yes, we will fix this as part of DataSense tests.

**Comment by gsfernandes** [comment]
Jenkins validate

**Pull request merged**